src/main/java/nextstep/courses/domain/session/enrollment/Enrollment.java

```diff
@@ -0,0 +1,35 @@
+package nextstep.courses.domain.session.enrollment;
+
+import nextstep.courses.domain.session.Session;
+import nextstep.courses.exception.session.InvalidSessionStateException;
+import nextstep.courses.type.SessionState;
+import nextstep.courses.type.SessionType;
+import nextstep.payments.domain.Payment;
+import nextstep.users.domain.NsUser;
+
+public interface Enrollment {
+
+    void enroll(NsUser student, Payment payment);
```

Comment:
> 일단은 1번으로 개발 해봤는데 리뷰어님은 주로 어떤 방식으로 하시는지 궁금합니다.

payment나 nsUser는 일회성으로 사용하고 있는데, session은 상태로 관리하고 있네요!
어떠한 이유에서 상태로 관리하고 있나요? 🤔

저라면 이렇게 할것 같아요!

```suggestion
    void enroll(Session session, NsUser student, Payment payment);
```

Reply:
답변 감사합니다.

Session 을 넘겨서 처리하였습니다.